### DISCUSSION TITLE
Subfolders of "./rugby/build/Debug-iphonesimulator" are generated as "{POD_NAME}.framework", not "{POD_NAME}"

### DISCUSSION THREAD

**Discussion by ykkd** [swiftyfinch/Rugby#51] [Q&A] [answered]
Hi there, 
Apologize for asking question with few information.

[Problem]
I am trying to cache pods but generated folders under `./rugby/build/Debug-iphonesimulator` are all named "{POD_NAME}.framework".
Because of that, cached pods cannot be found when I build the project.

[Workaround]
Compiling succeeds when I put "{POD_NAME}.framework" into each folder named "{POD_NAME}".

I created new folders for each. and just put {POD_NAME}.framework folder in...
File path will be like this↓
-> `./rugby/build/Debug-iphonesimulator/{POD_NAME}/{POD_NAME}.framework`

The thing is, I have no idea how to fix this issue..
Can anyone help me in this?

Below is a part of `rawBuild.log`
```
Command line invocation:
    /Applications/Xcode12.5.app/Contents/Developer/usr/bin/xcodebuild -project Pods/Pods.xcodeproj -scheme RugbyPods -sdk iphonesimulator SYMROOT=/Users/{USER_NAME}/Documents/01Develop/{APP_NAME}/.rugby/build COMPILER_INDEX_STORE_ENABLE=NO SWIFT_COMPILATION_MODE=wholemodule ARCHS=x86_64

User defaults from command line:
    IDEPackageSupportUseBuiltinSCM = YES

Build settings from command line:
    ARCHS = x86_64
    COMPILER_INDEX_STORE_ENABLE = NO
    SDKROOT = iphonesimulator14.5
    SWIFT_COMPILATION_MODE = wholemodule
    SYMROOT = /Users/{USER_NAME}/Documents/01Develop/{APP_NAME}/.rugby/build

Build Preparation
Build task concurrency set to 0 via user default IDEBuildOperationMaxNumberOfConcurrentCompileTasks

note: Using new build system
note: Building targets in parallel
note: Planning build
note: Analyzing workspace
note: Constructing build description
note: Build preparation complete
CreateBuildDirectory /Users/{App_NAME}/Documents/01Develop/{APP_NAME}/.rugby/build
    cd /Users/{USER_NAME}/Documents/01Develop/{APP_NAME}/Pods/Pods.xcodeproj
    builtin-create-build-directory /Users/{USER_NAME}/Documents/01Develop/{APP_NAME}/.rugby/build

CreateBuildDirectory /Users/{USER_NAME}/Documents/01Develop/{APP_NAME}/.rugby/build/Debug-iphonesimulator
    cd /Users/{USER_NAME}/Documents/01Develop/{APP_NAME}/Pods/Pods.xcodeproj
    builtin-create-build-directory /Users/{USER_NAME}/Documents/01Develop/{APP_NAME}/.rugby/build/Debug-iphonesimulator

=====
# CreateBuildDirectory for a folder named "Alamofire" should be executed here but not happening
=====

MkDir /Users/{USER_NAME}/Documents/01Develop/{APP_NAME}/.rugby/build/Debug-iphonesimulator/Alamofire.framework (in target 'Alamofire' from project 'Pods')
    cd /Users/{USER_NAME}/Documents/01Develop/{APP_NAME}/Pods
    /bin/mkdir -p /Users/{USER_NAME}/Documents/01Develop/{APP_NAME}/.rugby/build/Debug-iphonesimulator/Alamofire.framework

...
```

Thanks in advance.

**Comment by swiftyfinch** [comment, marked as answer]
Hello! I'll try to investigate this case on the weekend.
Can you share a test project for reproducing this issue?
Feel free to add more information if you get some earlier than me.

**Reply by ykkd** [reply]
thank you for your reply!

> Can you share a test project for reproducing this issue?

I am working on it but maybe it'll take some time.

Perhaps I can send a pull request for this.
I will post update!

**Reply by ykkd** [reply]
I fixed folder path issue at [this branch](https://github.com/ykkd/Rugby/tree/fix/cached_pods_path)

However, I found that I had one more issue.
generated info.plist of each framework don't have "CFBundleIdentifier".
<img width="863" alt="genereted_plist" src="https://user-images.githubusercontent.com/17854586/136615336-e241f5ca-394c-45df-ad6b-57ca31d347d5.png">

because of that, simulator will not be launched while compiling succeeds.

**Reply by ykkd** [reply]
@swiftyfinch 
Hi, I managed to reproduce this issue.

Please take a look at [this branch](https://github.com/ykkd/Pokedex/tree/feature/rugby_fails)
If I ran rugby cache step, the subfolder issue appears & CFBundleIdentifier is missing in generated framework.

On the other hand, [this branch](https://github.com/ykkd/Pokedex/tree/feature/rugby) has no issue integrating rugby.
I guess one of the project settings defined in `.xccconfig` files is the cause of this issue? 🤔

**Reply by swiftyfinch** [reply]
@ykkd Thank you for sharing more info 👍🏻

The problem is 🏈 `Rugby` build **Debug** config by default.
So, you renamed your **Debug** config to **Debug_Staging** and `Rugby` doesn't know about that.

If we pass concrete config name everything would be okay:

<img width="672" alt="Screenshot 2021-10-09 at 15 46 06" src="https://user-images.githubusercontent.com/64660122/136658492-b3cbd928-b9ae-40dd-936f-55c18814dd18.png">
<img width="374" alt="Screenshot 2021-10-09 at 15 45 24" src="https://user-images.githubusercontent.com/64660122/136658514-7a4254b0-25ff-4ffd-9c52-154072dc1815.png">

Do you want to make a pull request and add the ability to pass config name to 🏈 `Rugby`?
Or as a quick workaround, you can keep **Debug** config in your project.

**Reply by ykkd** [reply]
@swiftyfinch 
Thank you!
I understood the cause of this problem.

> Do you want to make a pull request and add the ability to pass config name to 🏈 Rugby?

Definitely💪
My current understanding is that 
- Build config name should be passed manually (as Option)
- Default value should be 'Debug'
- Update Cache.md as well

Anyways, I'll send a PR about this soon.
thank you for helping me out!

**Reply by swiftyfinch** [reply]
#52 Pull Request from @ykkd

**Reply by swiftyfinch** [reply]
@ykkd Hey, there is something strange with your GitHub account.
I don't see you in contributors 🙁
Probably, it's GitHub issue, you can ask them a question about that.

Anyway, thanks a lot for your contribution! 🏈
